lib/src/impl/realtime/connection.dart

```diff
@@ -15,6 +15,10 @@ class ConnectionPlatformObject extends PlatformObject implements Connection {
     this.handle;  //proactively acquiring handle
     this.state = ConnectionState.initialized;
     this.on().listen((ConnectionStateChange event) {
+      if (event.reason?.code == ErrorCodes.authCallbackFailure) {
```

Comment:
I see, here `ErrorCodes` is used as `int`. Is there a way to unify this to `int` or `String`?

Reply:
@zoechi Not sure! As per the spec, `ErrorInfo#code` is supposed to be int, and `PlatformException#code` is of type string. Not sure if we can unify them both.

@QuintinWillison Any thoughts?

Reply:
I'm not quite sure what's being proposed, by from my bird's eye view I think it's got to be lowest common denominator, surely? You can safely represent an `int` in a `String`, but not vice-versa.

Reply:
I'd then suggest we leave it as it is, when there is no concrete reason to change it.
I just wanted to be sure nothing was missed.